src/kOS/Suffixed/OrbitInfo.cs

```diff
@@ -107,6 +108,12 @@ private OrbitInfo GetNextPatch()
             return ! GetHasNextPatch() ? null : new OrbitInfo(orbit.nextPatch,Shared);
         }
 
+
+        private double GetNextPatchETA()
+        {
+            return  ! GetHasNextPatch() ? Double.PositiveInfinity : orbit.EndUT - Planetarium.GetUniversalTime();
```

Comment:
I guess the question would be what happens if you keep the handle to the transition in memory, and then check `EndUT` after the transition.  That would be the only way that zero and negative values may be valid.  In fact, I'll want to test that anyways to see if KSP spouts out some silly value.  I'll report me results.

You can indeed throw an exception.  You can use `kOS.Safe.Exceptions.KOSSituationallyInvalidException`.  Just make sure to be clear in the text about the cause.  This will have the same effect as returning infinity, but with an error message that the user can actually use to correct their behavior.  I'm OK with throwing an exception because there is a way to detect the "error" case other than by the exception itself (via `HASNEXTPATCH`).